apps/web/components/dashboard-components/project-details/custom-llm-parameters/parameter-row.tsx

```diff
@@ -0,0 +1,209 @@
+import { useState, useEffect, useMemo } from "react";
+import { motion } from "framer-motion";
+import { Input } from "@/components/ui/input";
+import {
+  Select,
+  SelectTrigger,
+  SelectValue,
+  SelectContent,
+  SelectItem,
+} from "@/components/ui/select";
+import { Textarea } from "@/components/ui/textarea";
+import { Button } from "@/components/ui/button";
+import { Trash2 } from "lucide-react";
+import { validateValue, shouldUseTextarea } from "./utils";
+import { PARAMETER_SUGGESTIONS, type ParameterEntry } from "./types";
+
+/**
+ * ParameterRow Component
+ *
+ * Represents a single editable parameter entry in the LLM parameters editor.
+ * Provides inline editing capabilities with type-specific input controls and
+ * validation. Handles both custom parameters and pre-defined suggestions.
+ */
+interface ParameterRowProps {
+  index: number;
+  param: ParameterEntry;
+  isEditing: boolean;
+  onBeginEdit: (rowIndex: number) => void;
+  onRemoveRow: (rowIndex: number) => void;
+  onParameterChange: (index: number, updatedParam: ParameterEntry) => void;
+  allowCustomParameters?: boolean;
+}
+
+export function ParameterRow({
+  index,
+  param,
+  isEditing,
+  onBeginEdit,
+  onRemoveRow,
+  onParameterChange,
+  allowCustomParameters = true,
+}: ParameterRowProps) {
+  const [local, setLocal] = useState<ParameterEntry>(param);
+  const [validationError, setValidationError] = useState<string | null>(null);
+
+  // Reset local state when param changes from parent
+  useEffect(() => {
+    if (!isEditing) {
+      setLocal(param);
+      setValidationError(null);
+    }
+  }, [param, isEditing]);
+
+  const handleChange = (field: keyof ParameterEntry, value: string) => {
+    if (!isEditing) onBeginEdit(index);
+
+    let updatedParam = { ...local, [field]: value };
+    let error: string | null = null;
+
+    // Validate value changes
+    if (field === "value") {
```

Comment:
these should just be fundamentally different callbacks (handleKeyChange and handleValueChange)- don't use magic values ("key", "value") to distinguish the cases